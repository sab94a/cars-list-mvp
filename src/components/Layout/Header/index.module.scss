@import 'components/lib/variables.scss';

.root {
    display: flex;
    justify-content: space-between;
    border-bottom: $border;
}

.logo {
    display: flex;
    align-items: center;
    padding: 0 $offset-lg;
    height: $panel-height;
}