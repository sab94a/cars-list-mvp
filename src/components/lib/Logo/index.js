// @flow

import React from 'react';
import logo from './logo.png';
import styles from './index.module.scss';

const Logo = () => (
    <img className={ styles.root } alt="Logo" src={ logo } />
);

export default Logo;
