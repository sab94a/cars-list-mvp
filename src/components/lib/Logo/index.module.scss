@import 'components/lib/variables.scss';

.root {
    display: block;
    height: $panel-height - $offset-lg * 2;
    align-items: center;
}