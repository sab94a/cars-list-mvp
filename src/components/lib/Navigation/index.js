// @flow

import React from 'react';
import Link from  'components/lib/Link';
import type { NavItem } from 'components/lib/types';
import styles from './index.module.scss';

type Props = {
    items: Array<NavItem>
};

const Navigation = ({ items }: Props) => (
    <nav className={ styles.root }>
        { items.map(({ href, title }:NavItem, index:number) => (
            <Link key={ index } href={ href } className={ styles.link }>{ title }</Link>
        )) }
    </nav>
);

export default React.memo<Props>(Navigation);
