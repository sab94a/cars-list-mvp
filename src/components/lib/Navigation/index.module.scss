@import 'components/lib/variables.scss';

.root {
    display: flex;
    flex-direction: row;
    height: $panel-height;
}

.link {
    display: flex;
    align-items: center;
    font-size: $font-size-sm;
    padding: 0 $offset-lg;
    text-decoration: none;
}