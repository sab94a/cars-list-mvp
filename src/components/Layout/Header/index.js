//@flow

import React from 'react';
import { Link } from 'react-router-dom';
import Navigation from 'components/lib/Navigation';
import Logo from 'components/lib/Logo';
import type { NavItem } from 'components/lib/types';
import styles from './index.module.scss'

type Props = {
    navigation: Array<NavItem>
};

const Header = ({ navigation }: Props) => (
    <header className={ styles.root }>
        <Link className={ styles.logo } to='/'>
            <Logo />
        </Link>
        <Navigation items={ navigation } />
    </header>
);

export default React.memo<Props>(Header);
