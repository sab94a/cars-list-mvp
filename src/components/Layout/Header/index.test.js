import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Link } from 'react-router-dom';
import Navigation from 'components/lib/Navigation';
import Logo from 'components/lib/Logo';
import Header from './';

const navigation = [{ href: '', title: ''}];

describe('<Header>', () => {
    let component =  null

    beforeEach(() => {
        component = shallow(<Header navigation={ navigation } />)
    });

    it('Should render correctly', () => {
        expect(toJson(component)).toMatchSnapshot();
    });

    it('Should has Logo, Link and Navigation with right props', () => {
        const LogoComponent = component.find(Logo);
        const LinkComponent = component.find(Link);
        const NavigationComponent = component.find(Navigation);

        expect(LogoComponent).toHaveLength(1);
        expect(LinkComponent).toHaveLength(1);
        expect(NavigationComponent).toHaveLength(1);
        expect(NavigationComponent.prop('items')).toEqual(navigation);
    });
});
