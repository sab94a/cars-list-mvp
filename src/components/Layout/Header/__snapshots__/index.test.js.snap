// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header> Should render correctly 1`] = `
<header
  className="root"
>
  <Link
    className="logo"
    to="/"
  >
    <Logo />
  </Link>
  <Memo(Navigation)
    items={
      Array [
        Object {
          "href": "",
          "title": "",
        },
      ]
    }
  />
</header>
`;
